### PR TITLE
OCPNODE-1539: perf profile: add script for preparing cgroups for CPU load balance disabling

### PR DESCRIPTION
**Pull request by haircommander**
on RHEL 9 and newer kernels, the sysfs previously relied on to disable CPU load balancing is no longer available. Instead, cpu load balancing must be configured through cgroups.

To configure a cpuset to have load balance disabled (on cgroup v1), a cpuset cgroup must have `cpuset.sched_load_balance` set to 0 (disable), and any cpuset that contains the same set as `cpuset.cpus` must also have `cpuset.sched_load_balance` set to disabled.

To do this, we must move systemd and all host processes to a separate cpuset off of the root cgroup, and manually configure it to use the reserved cpus set by set_cpuaffinity. This won't have any effect on the actual processes themselves, but will open the opportunity for the root cgroup to have load balance disabled.

Also, preconfigure the machine.slice (which will later be created by podman).

This PR adds a script to do these things and a systemd unit to enable it, which is created when there are reserved cpus in the performance profile.

### CODE REVIEW

**Comment by browsell** [comment]
LGTM

**Comment by yanirq** [comment]
@haircommander  can you please remove all 3 occurrences of `testutils.KnownIssueJira("OCPNODE-1538")` from the e2e tests so the relevant tests will run ? 

These are skips that were added to unblock CI until the issue is fixed (with this PR)

**Comment by MarSik** [comment]
@yanirq This change is not enough to fix OCPNODE-1538, it must be accompanied by https://github.com/cri-o/cri-o/pull/6744 

So the removal of the KnownIssue marker will not make the test pass successfully.

**Comment by yanirq** [comment]
> @yanirq This change is not enough to fix OCPNODE-1538, it must be accompanied by [cri-o/cri-o#6744](https://github.com/cri-o/cri-o/pull/6744)
> 
> So the removal of the KnownIssue marker will not make the test pass successfully.

@MarSik  but we do want to remove the skip and only if the tests pass successfully then merge this PR.
We can hold this PR as well until [cri-o/cri-o#6744](https://github.com/cri-o/cri-o/pull/6744) is resolved

**Comment by Tal-or** [comment]
We need to add an e2e test for that. at least make sure the file is present and setup correctly.

File: pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go
Comment (Tal-or) [inline pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go:63]: You should add the script itself to the system using [addContent](https://github.com/openshift/cluster-node-tuning-operator/blob/3dd2403b1617e612a09f9f3b9ff20a6ce0035b80/pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go#L434)

**Review by Tal-or — COMMENTED** [review]
I would recommend adding at least sanity e2e just to make sure the script is present and the systemd unit setup correctly

**Comment by haircommander** [comment]
> @MarSik but we do want to remove the skip and only if the tests pass successfully then merge this PR.
We can hold this PR as well until https://github.com/cri-o/cri-o/pull/6744 is resolved 

agreed, we can use this as the final piece and merge kubelet, cri-o and runc pieces first

File: pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go
Comment (haircommander) [reply pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go]: I see I thought that was taken care of above but I was wrong, thanks!

File: assets/performanceprofile/scripts/cpuset-configure.sh
Comment (MarSik) [inline assets/performanceprofile/scripts/cpuset-configure.sh:10]: I think the commonly used name is "system.slice".

File: assets/performanceprofile/scripts/cpuset-configure.sh
Comment (MarSik) [inline assets/performanceprofile/scripts/cpuset-configure.sh:32]: How is podman using the machine slice? Is that a workload partitioning specialty?

File: assets/performanceprofile/scripts/cpuset-configure.sh
Comment (haircommander) [reply assets/performanceprofile/scripts/cpuset-configure.sh:10]: I specifically chose not to use `.slice` because slices are something unique to systemd and the name would imply it's being managed by systemd, which it's not

File: assets/performanceprofile/scripts/cpuset-configure.sh
Comment (haircommander) [reply assets/performanceprofile/scripts/cpuset-configure.sh:32]: podman puts all of its systemd cgroup containers into the machine.slice by default. it was formally not at all configured for workload partitioning because the podman pieces run before guaranteed pods are in the picture. The slice remains on the node only as a remnant because nothing removes them

File: assets/performanceprofile/scripts/cpuset-configure.sh
Comment (haircommander) [reply assets/performanceprofile/scripts/cpuset-configure.sh:10]: I specifically chose not to use `.slice` because slices are something unique to systemd and the name would imply it's being managed by systemd, which it's not

**Comment by MarSik** [comment]
I am ok with the patch in general, but I think I would prefer if the slices were created via the systemd mechanism (a .slice file in /etc/systemd/system) so it is obvious where they come from though. If that is possible of course. system.slice might be a reserved name even though systemd is not creating it by default for the cpuset controller.

**Comment by haircommander** [comment]
cri-o is out of date in 4.14 so the test won't pass yet

**Comment by haircommander** [comment]
testing 4.13 on https://github.com/openshift/cluster-node-tuning-operator/pull/617 which has the required versions

**Review by ffromani — APPROVED** [review]
/approve
/lgtm

discussed extensively offline + followups already planned

**Comment by yanirq** [comment]
/override ci/prow/e2e-gcp-pao-updating-profile

test is reaching full timeout but non related to this PR

**Pull request merged**
